leetcode/src/658.c

```diff
@@ -0,0 +1,31 @@
+/**
+ * Note: The returned array must be malloced, assume caller calls free().
+ */
+int compXClose(const int *i, const int *j, const int* x)
+{
+    return abs(*i - *x) - abs(*j - *x);
+}
+
+int stdcomp(const void * a, const void * b) //what is it returning?
+{
+   return ( *(int*)a - *(int*)b ); //What is a and b?

```

Comment:
Sad pointer manipulation 😓 It took me so long to understand what is happening here.

Reply:
Looks like I'm quite late:sweat_smile:, but here's a better `cmp`, I'm going to post it, for posterity's sake if nothing else, this PR can stay open awhile longer for all I care. [Source](https://stackoverflow.com/questions/27284185/how-does-the-compare-function-in-qsort-work/27284248#27284248). This PR really reminded me of how I once trolled my friends by challenging them to sort an array containing both `INT_MAX` and `INT_MIN` with `qsort()` 😂
```suggestion
   const int *A = (int*)a, *B = (int*)b;
   return (*A > *B) - (*A < *B);
```